pkg/pillar/cmd/nodeagent/handlebaseos.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// base os event handlers
+
+package nodeagent
+
+import (
+	"fmt"
+	"github.com/lf-edge/eve/pkg/pillar/cast"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+)
+
+func lookupBaseOsStatusOnPartition(ctx *nodeagentContext, key string) *types.BaseOsStatus {
+	sub := ctx.subBaseOsStatus
+	items := sub.GetAll()
+	for _, st := range items {
+		status := cast.CastBaseOsStatus(st)
+		if status.PartitionLabel == key {
+			return &status
+		}
+	}
+	return nil
+}
+
+// mark the zedcloud health/connectivity test complete flag
+// for baseosmgr to pick up and complete the partition activation
+func initiateBaseOsZedCloudTestComplete(ctx *nodeagentContext) {
+	if !ctx.updateInprogress {
+		return
+	}
+	log.Infof("initiateBaseOsZedCloudTestComplete(%s)\n", ctx.curPart)
+	// get the current partition zboot config and status
+	zbootConfig := lookupZbootConfig(ctx, ctx.curPart)
```

Comment:
Initially isn't zbootConfig absent? Where to do initially create it for the two partitions?

Reply:
the zboot configs are published inside nodeagent .Run() function.